source/controllers/PPGameStateController.cpp

```diff
@@ -2,8 +2,11 @@
 
 void GameStateController::_jsonv1_loadColors(const json_t &colors) {
     _state.colors.clear();
-    for (const auto &n : colors->asArray()) {
-        const vec<int> c = n->asIntArray();
+    string shapes[] = {"color-circle", "color-heart", "color-square", "color-star", "color-triangle"};
```

Comment:
I'm not sure if this is a better or worse option, but maybe it's worth setting the mapping of a color to shape in a JSON? I'm concerned that if a level generation is messed up where the ordering of colors is not the same as before the same color may be represented by different shapes in different levels, which may confuse some. Of course I'm also not sure if this will work if we add colors besides the ones we currently have, but it may be worth considering.

Reply:
Yes, it is very possible that different shapes can be mapped to different colors across levels, but if this on the other hand, the order of the shapes will always remain consistent. We can always set the configuration mapping color indices to shapes in a global json too, but I didn't want to modify the structure of that JSON just yet (+ it would require changing code in other files in an already-large PR)

Reply:
I mean, if they colorblind, they gonna rely on the shapes... which as long as within the level it's consistent I don't see why it matters

Reply:
I'm not sure that's necessarily true. Most people aren't truly colorblind, they just can't differentiate between like red and green or red and blue. So they may be able to distinguish some colors, but not all. But honestly for this PR I agree with Ansh that it's not worth doing rn, especially in this thing, but it's likely something we need to keep in mind, especially as levels continue to get created. It also may not be worth it for all colors, but if we always have a red chosen, it would be nice to have red consistently be the same shape then.

Reply:
Actually, (this is coming from a Slack conversation with Leo), I don't think reds should always map to the same shape because we need to have a certain level of randomization anyway. Having different mappings is one of the only ways the player doesn't memorize the level completely (since we've decided against randomizing each play of the same level)

Reply:
That is a good point, then this is fine by me. I see no issues here/I think we're good to merge.